decompiler/structures/pseudo/typing.py

```diff
@@ -18,34 +20,22 @@ def is_boolean(self) -> bool:
         """Check whether the given value is a boolean."""
         return str(self) == "bool"
 
-    def copy(self, **kwargs) -> Type:
-        """Generate a copy of the current type."""
-        return replace(self, **kwargs)
-
-    def resize(self, new_size: int) -> Type:
+    def resize(self: _T, new_size: int) -> _T:
         """Create an object of the type with a different size."""
-        return self.copy(size=new_size)
+        return replace(self, size=new_size)
 
     @abstractmethod
     def __str__(self) -> str:
         """Every type should provide a c-like string representation."""
 
-    def __add__(self, other) -> Type:
-        """Add two types to generate one type of bigger size."""
-        return self.copy(size=self.size + other.size)
-
-    def __hash__(self) -> int:
-        """Return a hash value for the given type."""
-        return hash(repr(self))
-
 
 @dataclass(frozen=True, order=True)
 class UnknownType(Type):
     """Represent an unknown type, mostly utilized for testing purposes."""
 
     def __init__(self, size: int = 0):
         """Create a type with size 0."""
-        super().__init__(size)
+        object.__setattr__(self, "size", size)
```

Comment:
Calling init for the super type of a dataclass only works by accident. Normally, each dataclass is responsible for initializing all fields, including inherited fields from super types. I was testing some stuff with dataclasses and ran into issues because of that.

Reply:
Okay, thank you! I wasn't aware of that issue.